Automattic/wp-calypso
# TextDiff: Decouple TextDiff component from post revisions

### Summary
This PR is an effort towards decoupling some components that have been built during the development of the post revisions work, making them reusable in other areas of the project, easier to unit test and easier to document.

Specifically - here we're moving out the component that shows additions and removals to text/content in to a new component: `TextDiff`.

### Testing
There should be no deviations from the original functionality in post revisions.
To test this:
- Go to a the post editor
- Make some revisions if the post doesn't already have a few
- Click on the 'history' button in the editors header
- You should see removals highlighted in a faint red, additions in a faint blue exactly as before
- Also pay attention to the line splitting, the title shouldn't break changes on to new lines but the changes in the content should.

Also, take a look at the DevDocs entry - does this look reasonable? Is it worth showing multiple examples? 
Go to https://calypso.live/devdocs and search TextDiff in 'UI Components'.

### TODO:
- [x] Add readme for `TextDiff`
- [x] Add `TextDiff` to DevDocs
- [ ] Potentially rename - what do you think?
